### PR TITLE
implement Base.complex, Base.widen, Base.reim

### PR DESCRIPTION
**Pull request by steinQF**
This PR helps dealing with complex, Unitful quantities by implementing missing methods.

Complex and yet Unitful quantities do occur e.g. as generalized frequencies or in the Ray Transfer matrix treatment of Gaussian beams.

### CODE REVIEW

**Comment by steinQF** [comment]
I'm surprised at the two test failures (appveyor x86 for Julia v0.7 and v1.0 but not for x64), especially since one of them failed in only 17 seconds and the another one timed out. Is it possible that these failures are unrelated to the code, perhaps due to some networking problem?

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@
 deps/
 docs/build/
 docs/site/
+*~
```
Comment (giordano) [inline .gitignore]: Please *always* end a file with a newline

Comment (steinQF) [reply .gitignore]: Yes, sorry, I'll change that.

**Comment by steinQF** [comment]
I'm abandoning this PR as testing revealed that the `Base.reim` method has a bug. I'll post a new PR when that is fixed.

**Pull request closed without merging**